test/test_examples.py

```diff
@@ -1112,6 +1112,29 @@ def test_kl_div(self):
             )
         )
 
+    @skipIfRocm("failure on rocm")
```

Comment:
Instead of using `skipIfRocm` which skips the whole test including the output equality check, maybe we can add a `skip_rocm: bool` arg to `def assertExpectedJournal` that skips the journal check if the device is rocm. And to check the device is rocm, we can add something like this to `_testing.py`:
```python
def is_rocm() -> bool:
    """Return True if running on ROCm (AMD GPU)."""
    return (
        triton.runtime.driver.active.get_current_target().backend == "hip"
        and DEVICE.type == "cuda"
    )
```
(Please feel free to do this in a follow-up PR. Thanks!)